tests/network/vmi_istio.go

```diff
@@ -155,6 +155,24 @@ var istioTests = func(vmType VmType) {
 			By("Waiting for VMI to be ready")
 			tests.WaitUntilVMIReady(vmi, console.LoginToAlpine)
 		})
+		Describe("Sysctl route_localnet configuration", func() {
+			BeforeEach(func() {
+				if vmType == Passt {
+					Skip("route_localnet is not relevant for passt")
+				}
+			})
+			It("Should have route_localnet enabled for in-pod bridge", func() {
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, util.NamespaceTestDefault)
+				podOutput, err := tests.ExecuteCommandOnPod(
+					virtClient,
+					vmiPod,
+					"compute",
+					[]string{"cat", "/proc/sys/net/ipv4/conf/k6t-eth0/route_localnet"},
+				)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(podOutput).To(ContainSubstring("1"), "should have sysctl route_localnet enabled")
+			})
+		})
```

Comment:
Hm I actually meant a test where "traffic works" - imo, that is what an e2e test would look like.

... but I understand that would be impossible right now because the cluster would have to work with ovn-k as the default CNI. 

Would you  document in the test what this knob - the `route_localnet` is used for ? As you've already done for the previous commit.

Reply:
Yeah, I understand your point, the problem is that we already test that "traffic works", but that's only because this setting is already enabled by default.

>Would you document in the test what this knob - the route_localnet is used for ? As you've already done for the previous commit.

Of course

Reply:
[Done](https://github.com/kubevirt/kubevirt/compare/80d4dd979c35e1832ba5e656114289b61a28a548..9f46454b4ea2cba89641cd08120cb5c720811e91)

Reply:
Sorry to keep insisting, but would it make sense (honest question) to write the test in the following way ?
```golang
Context("a pod whose in-pod bridge features the `route_localnet` sysctl *disabled*", func() {
    It("cannot route traffic originating from the istio proxy", func(){
        ...
    })
    
    When("the in-pod bridge `route-localnet knob is enabled`", func(){
        It("becomes able to route traffic originating from the istio proxy", func(){
            ...
        })        
    })
})
```

Probably this way you can avoid the comment. 

Anyway, as I've said, I'm OK with the shape of the code - let me know your thoughts, so I can add the lgrm label now, or wait for more changes.

Reply:
hm, after thinking about this 5 seconds longer, I don't think you can actually write a test like this: virt-handler is **always** pushing this button, thus you can't enable / disable it. 

... would have been nice though,

Reply:
I tried to pursue this, I thought that I could have a setup that would set the sysctl route_localnet option to 0, because
a) as you say, it's currently always enabled in the handler, 
b) our kubevirtci setup has this enabled on `all` interfaces, so in order to test it, it needs to be disabled for `all` as well
So we would need something like this
```go
JustBeforeEach(func() {
	for _, iface := range []string{"all", "k6t-eth0"} {
		vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, util.NamespaceTestDefault)
		_, err := tests.ExecuteCommandOnPod(
			virtClient,
			vmiPod,
			"compute",
			[]string{"/usr/bin/bash", "-c", fmt.Sprintf("echo 0 > /proc/sys/net/ipv4/conf/%s/route_localnet", iface)},
		)
		Expect(err).NotTo(HaveOccurred())
	}
})
```
But I can't write to the file because I get
`/proc/sys/net/ipv4/conf/all/route_localnet: Read-only file system`

Reply:
IMO, that's expected - you're accessing the launcher pod (it is where the VMI is executed ...) and trying to sysctl from there.

You need special privileged to do that; I think you can do this via virt-handler, and tapping into that launcher's netns.

I don't think it makes much sense, let's leave your test as is.